keanu-project/src/test/java/io/improbable/keanu/network/BayesianNetworkTest.java

```diff
@@ -141,4 +144,22 @@ public void testGetAverageVertexDegree() {
         assertThat(network.getAverageVertexDegree(), equalTo((1. + 1. + 2. + 2. + 2.) / 5));
     }
 
+    @Test
+    public void networkReturnsVerticesInNamespace() {
+        BooleanVertex a0 = new BernoulliVertex(0.5);
+        BooleanVertex a1 = new BernoulliVertex(0.5);
+        BooleanVertex b0 = new BernoulliVertex(0.5);
+        BooleanVertex c = new BernoulliVertex(0.5);
+
+        a0.setLabel(new VertexLabel("0", "root", "a"));
+        a1.setLabel(new VertexLabel("1", "root", "a"));
+        b0.setLabel(new VertexLabel("0", "root", "b"));
+
+        BayesianNetwork net = new BayesianNetwork(Arrays.asList(a0, a1, b0, c));
+        List<Vertex> verticesInNamespace = net.getVerticesInNamespace("root");
```

Comment:
Can you try a test that has labels you don’t want to include?

Reply:
You mean vertices that don't have the label specified in the argument of `getVerticesInNamespace`? Vertex c isn't returned here